### PR TITLE
[#3815]Improvement(lakehouse-iceberg): Add docs about authentication configuration for Iceberg catalog.

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Add documents about the configuration items for the Iceberg catalog.

### Why are the changes needed?

For ease of use for users. 

Fix: #3815

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

N/A

### CODE REVIEW

File: docs/lakehouse-iceberg-catalog.md [hunk 1]
```diff
@@ -33,11 +33,18 @@ Builds with Hadoop 2.10.x, there may be compatibility issues when accessing Hado
...
+| `uri`                                              | The URI configuration of the Iceberg catalog. `thrift://127.0.0.1:9083` or `jdbc:postgresql://127.0.0.1:5432/db_name` or `jdbc:mysql://127.0.0.1:3306/metastore_db` or `http://127.0.0.1:9001`. | (none)        | Yes                                                         | 0.2.0         |
+| `warehouse`                                        | Warehouse directory of catalog. `file:///user/hive/warehouse-hive/` for local fs or `hdfs://namespace/hdfs/path` for HDFS.                                                                      | (none)        | Yes                                                         | 0.2.0         |
+| `authentication.type`                              | The type of authentication for Iceberg catalog backend, currently we only support `kerberos`, `simple`.                                                                                         | `simple`      | No                                                          | 0.6.0         |
```
Comment (qqqttt123) [inline docs/lakehouse-iceberg-catalog.md:41]: Do we have this config option? Hive doesn't have this property now.

Comment (yuqi1129) [reply docs/lakehouse-iceberg-catalog.md:41]: I have added this to the Iceberg Hive backend configuration item.

Comment (qqqttt123) [reply docs/lakehouse-iceberg-catalog.md:41]: Could you align to the config option of Hive?

File: docs/lakehouse-iceberg-catalog.md [hunk 2]
```diff
@@ -33,11 +33,18 @@ Builds with Hadoop 2.10.x, there may be compatibility issues when accessing Hado
...
+| `authentication.impersonation-enable`              | Whether to enable impersonation for the Hadoop catalog backend.                                                                                                                                 | `false`       | No                                                          | 0.6.0         |
```
Comment (qqqttt123) [inline docs/lakehouse-iceberg-catalog.md:42]: Why is this Hadoop catalog?

File: docs/lakehouse-iceberg-catalog.md (hunk 1 above)
Comment (yuqi1129) [reply docs/lakehouse-iceberg-catalog.md:41]: Sure, let me check whether we need to check Hive or Iceberg for this point to align the configuration items.

Comment (qqqttt123) [reply docs/lakehouse-iceberg-catalog.md:41]: Why is Hive special?

Comment (yuqi1129) [reply docs/lakehouse-iceberg-catalog.md:41]: We haven't added the configuration for `Hive` yet. If it's necessary, I'll add it.

Comment (qqqttt123) [inline docs/lakehouse-iceberg-catalog.md:41]: kerberos -> Kerberos?

Comment (qqqttt123) [inline docs/lakehouse-iceberg-catalog.md:41]: Could you avoid `we`?

**Comment by qqqttt123** [comment]
Does the MySQL backend have authentication? Have we considered it?

**Comment by yuqi1129** [comment]
> Does the MySQL backend have authentication? Have we considered it?

We have used the username and password authentication mechanism. The only issue is that the configuration items can be used to cover JDBC authentication, as this PR is only for the Hive backend. 

For MySQL, authentication is almost necessary, so the item can be: 
```
authenticatin.type = username-password
authentication.username-password.user = xxxx
authentication.username-password.password = xxxx
```

The core issues lie in making it compatible with the previous configuration key.

**Review by qqqttt123 — APPROVED** [review]
LGTM.

**Pull request merged**